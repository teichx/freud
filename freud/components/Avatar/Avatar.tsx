import { forwardRef } from 'react';

import Image from 'next/image';

import { ChakraUiAvatar } from './style';
import { AvatarProps } from './types';

export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(
  function AvatarFunction({ name, src, w, h, ...props }, ref) {
    return (
      <ChakraUiAvatar
        {...props}
        ref={ref}
        name={name}
        position='relative'
        w={w ? `${w}px` : undefined}
        h={h ? `${h}px` : undefined}
      >
        <Image src={src} alt={src} width={w} height={h} />
      </ChakraUiAvatar>
    );
  }
);
