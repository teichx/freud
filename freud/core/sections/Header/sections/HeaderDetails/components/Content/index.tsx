import { ContentAppearance } from './ContentAppearance';
import { ContentDefault } from './ContentDefault';
import { ContentLanguage } from './ContentLanguage';
import { ContentProps, ContentVariationProps } from './types';

const contentVariation: ContentVariationProps = {
  close: () => null,
  default: ContentDefault,
  language: ContentLanguage,
  appearance: ContentAppearance,
};

export const Content = ({ toContent, content }: ContentProps) => {
  const ContentComponent = contentVariation[content];

  return <ContentComponent toContent={toContent} />;
};
