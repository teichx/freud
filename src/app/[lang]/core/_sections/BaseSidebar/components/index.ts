export * from './ButtonToggleSidebar';
export * from './SidebarButton';
export * from './SidebarWrapper';
