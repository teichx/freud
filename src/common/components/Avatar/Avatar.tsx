import { forwardRef } from 'react';

import { Avatar as ChakraUiAvatar } from '@chakra-ui/react';
import Image from 'next/image';

import { AvatarProps } from './types';

export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(
  function AvatarFunction({ name, src, w, h, ...props }, ref) {
    return (
      <ChakraUiAvatar
        sx={{
          borderRadius: '50%',
          img: {
            top: 0,
            left: 0,
            position: 'absolute',
            borderRadius: '50%',
          },
          ...props.sx,
        }}
        {...props}
        ref={ref}
        name={name}
        w={w ? `${w}px` : undefined}
        minW={w ? `${w}px` : undefined}
        h={h ? `${h}px` : undefined}
        minH={h ? `${h}px` : undefined}
      >
        <Image src={src} alt={src} width={w} height={h} />
      </ChakraUiAvatar>
    );
  }
);
