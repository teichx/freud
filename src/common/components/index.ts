export * from './Avatar';
export * from './Breadcrumb';
export * from './Buttons';
export * from './DataModal';
export * from './DataModal';
export * from './DataTable';
export * from './Display';
export * from './Form';
export * from './Input';
export * from './GoogleLogin';
export * from './InfiniteScroll';
export * from './Loader';
export * from './Logo';
export * from './Meta';
export * from './ReadOnlyText';
export * from './Section';
export * from './Select';
export * from './SkeletonContextLoader';
export * from './TextHighlight';
export * from './ToggleTheme';
export * from './TooltipComponent';
export * from './TooltipDate';
