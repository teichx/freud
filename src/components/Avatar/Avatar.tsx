import { forwardRef } from 'react';

import Image from 'next/image';

import { ChakraUiAvatar } from './style';
import { AvatarProps } from './types';

export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(
  function AvatarFunction({ name, src, w, h, ...props }, ref) {
    return (
      <ChakraUiAvatar
        {...props}
        ref={ref}
        name={name}
        w={w ? `${w}px` : undefined}
        minW={w ? `${w}px` : undefined}
        h={h ? `${h}px` : undefined}
        minH={h ? `${h}px` : undefined}
      >
        <Image src={src} alt={src} width={w} height={h} />
      </ChakraUiAvatar>
    );
  }
);
