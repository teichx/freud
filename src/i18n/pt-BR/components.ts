export const components = {
  buttons: {
    microphone: {
      toDisable: 'Desligar microfone',
      toEnable: 'Ativar microfone',
      helperText: {
        editOnlyNotListening:
          'O campo não pode ser editado enquanto o microfone estiver ligado',
      },
    },
    login: 'Fazer login',
    access: 'Entrar como {name}',
    logout: 'Deslogar',
    back: 'Voltar',
    cancel: 'Cancelar',
    save: 'Salvar',
  },
  toggleTheme: {
    label: 'Usar tema {appearance}',
    light: 'claro',
    dark: 'escuro',
  },
  form: {
    disabledByPristine: 'Não há alterações a serem salvas',
    disabledByHasValidationErrors: 'Há dados inválidos no formulário',
  },
  dataTable: {
    noData: 'Sem dados por aqui',
    loading: 'Carregando linhas',
    pagination: 'Página {page} de {pageCount}',
    rowsPerPage: 'Linhas por página:',
    pageHint: {
      first: 'Primeira página',
      next: 'Próxima página',
      before: 'Página anterior',
      last: 'Última página',
    },
  },
  select: {
    empty: {
      unfilled: 'Sem mais items',
      filled: 'Não há outros resultados para "{inputValue}"',
    },
    placeholder: {
      unique: 'Selecione uma opção',
      multiple: 'Selecione as opções',
    },
  },
  infiniteScroll: {
    endOfList: 'Sem mais items',
  },
  sidebar: {
    open: 'Abrir barra lateral',
    close: 'Fechar barra lateral',
  },
} as const;
