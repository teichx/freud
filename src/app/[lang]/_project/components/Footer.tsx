'use server';
import {
  Box,
  Container,
  Icon,
  IconButton,
  Text,
  HStack,
} from '@chakra-ui/react';
import { BsGithub } from 'react-icons/bs';
import { GiBrazil } from 'react-icons/gi';

import { LinkButton } from '~/common/components/Buttons/variant/LinkButton';
import { TooltipComponent } from '~/common/components/TooltipComponent';
import { getScopedI18n } from '~/i18n/server';

export const Footer = async () => {
  const t = await getScopedI18n('project');

  return (
    <Box
      w='100%'
      sx={{
        bg: 'book.darkBlue.500',
        color: 'gray.200',
        _dark: {
          bg: 'book.desertSun.900',
        },
      }}
    >
      <Container w='100%' maxW='container.xl' fontSize='sm'>
        <HStack
          justifyContent='center'
          py='2'
          columnGap='1'
          sx={{
            '> *': {
              marginInlineStart: '0 !important',
            },
          }}
        >
          <Text m='0'>{t('author.createdOn')}</Text>

          <TooltipComponent label={t('author.brazil')}>
            <Box
              display='flex'
              alignItems='center'
              sx={{
                color: 'book.desertSun.500',
                _dark: {
                  color: 'white',
                },
              }}
            >
              <Icon viewBox='0 0 14 14'>
                <GiBrazil />
              </Icon>
            </Box>
          </TooltipComponent>

          <Text>{t('author.by')}</Text>

          <LinkButton
            p='0'
            height='auto'
            color='inherit'
            variant='link'
            fontSize='inherit'
            fontWeight='inherit'
            linkProps={{
              target: '_blank',
            }}
            href='https://github.com/teichx'
          >
            {t('author.teichx')}
          </LinkButton>

          <IconButton
            size='xs'
            rounded='full'
            sx={{
              color: 'black',
              _dark: {
                color: 'white',
              },
            }}
            aria-label='https://github.com/teichx/freud'
            icon={
              <Icon viewBox='0 0 16 16' fontSize='md'>
                <BsGithub />
              </Icon>
            }
          />
        </HStack>
      </Container>
    </Box>
  );
};
