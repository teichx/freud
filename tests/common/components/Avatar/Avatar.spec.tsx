import { render } from 'test-utils';

import { Avatar } from '~/common/components/Avatar';

describe('component.Avatar', () => {
  it('renders with width and height', () => {
    expect.assertions(1);

    const { asFragment } = render(
      <Avatar src='http://great-url.com' alt='Great alt' w={40} h={40} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without width and height', () => {
    expect.assertions(1);

    const { asFragment } = render(
      <Avatar src='http://great-url.com' alt='Other alt' w={0} h={0} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
