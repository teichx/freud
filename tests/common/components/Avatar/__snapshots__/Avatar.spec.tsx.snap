// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.Avatar renders with width and height 1`] = `
<DocumentFragment>
  <span
    alt="Great alt"
    class="chakra-avatar css-11zvsst"
  >
    <svg
      aria-label=" avatar"
      class="chakra-avatar__svg css-16ite8i"
      role="img"
      viewBox="0 0 128 128"
    >
      <path
        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
        fill="currentColor"
      />
      <path
        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
        fill="currentColor"
      />
    </svg>
    <img
      alt="http://great-url.com"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fgreat-url.com&w=96&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fgreat-url.com&w=48&q=75 1x, /_next/image?url=http%3A%2F%2Fgreat-url.com&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </span>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`component.Avatar renders without width and height 1`] = `
<DocumentFragment>
  <span
    alt="Other alt"
    class="chakra-avatar css-wj3r0b"
  >
    <svg
      aria-label=" avatar"
      class="chakra-avatar__svg css-16ite8i"
      role="img"
      viewBox="0 0 128 128"
    >
      <path
        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
        fill="currentColor"
      />
      <path
        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
        fill="currentColor"
      />
    </svg>
    <img
      alt="http://great-url.com"
      data-nimg="1"
      decoding="async"
      height="0"
      loading="lazy"
      src="/_next/image?url=http%3A%2F%2Fgreat-url.com&w=16&q=75"
      srcset="/_next/image?url=http%3A%2F%2Fgreat-url.com&w=16&q=75 1x"
      style="color: transparent;"
      width="0"
    />
  </span>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
